firestore/docref.go

```diff
@@ -795,7 +795,8 @@ type DocumentSnapshotIterator struct {
 // the current state of the document. If the document has been deleted, Next
 // returns a DocumentSnapshot whose Exists method returns false.
 //
-// Next never returns iterator.Done unless it is called after Stop.
+// Next is not expected to return iterator.Done unless it is called after Stop.
```

Comment:
I am reluctant to add this because the reported issue was only for `QuerySnapshotIterator`. I have no idea whether this can also happen for this iterator or if the project is specific to QuerySnapshots.

Reply:
If one path is having this, the other one is likely having it as well. Watch is implemented to setup a WatchStream with a target, which is a bit generic. The underlying stream they iterate on is the 'same' and they both are permuting io.EOF to iterator.Done

Reply:
Ah yeah, I took another look and that seems correct. Makes sense.